### PR TITLE
Migration: Share modal 

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This PR replaces angular-based share modal dialog with a new one implemented in React. The main goal in this state is just to replace old modal window with new with no visual or behavior changes. However, there're some stuff can be revamped and redesigned in next iteration, I think.

One of the tricky part of this work is open modal window from panel menu or with hotkeys, because it's implemented with events. So I added new `CoreEvents.showModalReact` event which able to mount and show modal react component and then clean up and remove it from DOM.

Also, some code used angular features (mostly, working with URLs), so I had to reimplement it in pure typescript.

Any feedback appreciated.

### CODE REVIEW

**Comment by dprokop** [comment]
haha, similar: https://github.com/grafana/grafana/pull/22395

File: public/app/core/services/util_srv.ts
Comment (dprokop) [inline public/app/core/services/util_srv.ts]: I need this: https://github.com/grafana/grafana/pull/22395#discussion_r384372501 :)

Comment (alexanderzobnin) [reply public/app/core/services/util_srv.ts]: Yea, that was tricky. I also got some strange theme hadnling until I wraped component with `provideTheme()`. I'm not sure about wrapping with `WrapInProvider()`, because right now I don't need any state from redux store.

File: public/app/features/dashboard/components/ShareModal/ShareModal.tsx
Comment (torkelo) [inline public/app/features/dashboard/components/ShareModal/ShareModal.tsx]: This should be part of the Modal component somehow, maybe a namespace component on Modal like `Modal.TabsHeader title="asd" icon="share-square-o"><TabsBar>...` So the bascially a Modal.TabsHeader componet that captures the styling you have here for modalHeaderTitle etc .

**Review by torkelo — COMMENTED** [review]
Great job!  looks really good. Need to test it manually, but have scanned through the code once, and looks solid. Make sure the e2e runs (run them locally) as the e2e open this modal and click on the image render link.

Next step would be to use new form components but I think it makes sense to do that in another PR late (we can do that), to keep the scope on this smaller

File: public/app/features/dashboard/components/ShareModal/ShareModal.tsx
Comment (alexanderzobnin) [reply public/app/features/dashboard/components/ShareModal/ShareModal.tsx]: Yes, sounds good

**Comment by alexanderzobnin** [comment]
I'm struggling with tests, so will figure out with e2e right after fixing unit tests.

**Comment by dprokop** [comment]
@alexanderzobnin I've cherrypicked your work on the react modals being opened with an event and I have applied it here: https://github.com/grafana/grafana/pull/22395/commits/af4b502feb0e7bf7936e7598a85a5c53b343d8ea

Please note that React modals are rendered with AngularModalProxy component so that react modals can open another modals and automatically dismiss the proviously opened one. It's handled by ModalsContext https://github.com/grafana/grafana/pull/22395/commits/84556666324a5d686e02efa6b8de192480e432e3 which basically renders always the component that is currently in the context.

**Comment by alexanderzobnin** [comment]
@dprokop Ok, so I guess, it's better to wait for merging your PR first and then I can rebase this one.

File: public/app/features/dashboard/components/ShareModal/ShareModal.tsx
Comment (torkelo) [inline public/app/features/dashboard/components/ShareModal/ShareModal.tsx]: for core components you can just get theme from config.theme instead, no need to use with withTheme HOC

**Review by torkelo — APPROVED** [review]
Works and looks good, great job! 

Just the modal header bit that would be nice.

File: public/app/features/dashboard/components/ShareModal/ShareModal.tsx
Comment (alexanderzobnin) [reply public/app/features/dashboard/components/ShareModal/ShareModal.tsx]: Ok, sounds good.

**Comment by alexanderzobnin** [comment]
@torkelo I still need some tests for sharing dashboard. Also, I think, it's better to wait for #22395 being merged first since we're touching the same code.

File: public/app/features/dashboard/components/ShareModal/ShareModal.tsx
Comment (alexanderzobnin) [reply public/app/features/dashboard/components/ShareModal/ShareModal.tsx]: Created `ModalTabsHeader` and `ModalHeader` components for reusing modal header. It's a good idea to move it under `Modal` namespace, but I think it's too much work that touches other components, so better do it after merging.

**Comment by dprokop** [comment]
@alexanderzobnin #22395 is now merged

File: public/app/core/services/util_srv.ts
Comment (alexanderzobnin) [inline public/app/core/services/util_srv.ts]: @dominik without `provideTheme()` wrapper modal will always be dark when you show it via event emitter.

**Pull request merged**
